src/windows/View.tsx

```diff
@@ -0,0 +1,371 @@
+                <RNW.FocusableWindows
+                    {...focusableViewProps}
+                >
+                    {content}
+                </RNW.FocusableWindows>
+            );
+        }
+
+        return content;
+    }
+
+    private _onFocusableRef = (btn: RNW.FocusableWindows): void => {
+        this._focusableElement = btn;
+    }
+
+    focus() {
+        super.focus();
+        // Only forward to Button.
+        // The other cases are RN.View based elements with no meaningful focus support
+        if (this._focusableElement) {
+            this._focusableElement.focus();
+        }
+    }
+
+    blur() {
+        super.blur();
+        // Only forward to Button.
+        // The other cases are RN.View based elements with no meaningful focus support
+        if (this._focusableElement) {
+            this._focusableElement.blur();
+        }
+    }
+
+    getChildContext() {
+        // Let descendant Types components know that their nearest Types ancestor is not an Types.Text.
+        // Because they're in an Types.View, they should use their normal styling rather than their
+        // special styling for appearing inline with text.
+        let childContext: ViewContext = {
+            isRxParentAText: false
+        };
+
+        // Provide the descendants with the focus manager (if any).
+        if (this._focusManager) {
+            childContext.focusManager = this._focusManager;
+        }
+
+        return childContext;
+    }
+
+    setFocusRestricted(restricted: boolean) {
+        if (!this._focusManager || !this.props.restrictFocusWithin) {
+            console.error('View: setFocusRestricted method requires restrictFocusWithin property to be set to true');
+            return;
+        }
+
+        if (restricted) {
+            this._focusManager.restrictFocusWithin();
+        } else {
+            this._focusManager.removeFocusRestriction();
+        }
+    }
+
+    setFocusLimited(limited: boolean) {
+        if (!this._focusManager || !this.props.limitFocusWithin) {
+            console.error('View: setFocusLimited method requires limitFocusWithin property to be set to true');
+            return;
+        }
+
+        if (limited && !this._isFocusLimited) {
+            this._isFocusLimited = true;
+            this._focusManager.limitFocusWithin();
+        } else if (!limited && this._isFocusLimited) {
+            this._isFocusLimited = false;
+            this._focusManager.removeFocusLimitation();
+        }
+    }
+
+    private _onFocusableKeyDown = (e: React.SyntheticEvent<any>): void => {
+
+        let keyEvent = EventHelpers.toKeyboardEvent(e);
+        if (this.props.onKeyPress) {
+            this.props.onKeyPress(keyEvent);
+        }
+
+        if (this.props.onPress) {
+            let key = keyEvent.keyCode;
+            // ENTER triggers press on key down
+            if (key === KEY_CODE_ENTER) {
+                this.props.onPress(keyEvent);
+                return;
```

Comment:
>return [](start = 16, length = 6)

extra return

Reply:
ok

---
In reply to: [159359635](https://github.com/Microsoft/reactxp/pull/404#discussion_r159359635) [](ancestors = 159359635)